cmd/go-cache-plugin/commands.go

```diff
@@ -26,6 +26,7 @@ var flags struct {
 	CacheDir      string        `flag:"cache-dir,default=$GOCACHE_DIR,Local cache directory (required)"`
 	S3Bucket      string        `flag:"bucket,default=$GOCACHE_S3_BUCKET,S3 bucket name (required)"`
 	S3Region      string        `flag:"region,default=$GOCACHE_S3_REGION,S3 region"`
+	S3URL         string        `flag:"s3-url,default=$GOCACHE_S3_URL,S3 URL"`
```

Comment:
I think this flag needs a more descriptive name. Perhaps:
```suggestion
	S3Endpoint    string        `flag:"s3-endpoint-url,default=$GOCACHE_S3_ENDPOINT_URL,S3 custom endpoint URL (if unset, use AWS default)"`
```

In particular: Make the name reflect that this is the endpoint URL, and note the meaning of the default.